src/app/cli/src/app.rs

```diff
@@ -444,6 +473,12 @@ pub fn configure_cli_catalog(
 
 #[tracing::instrument(level = "info", skip_all)]
 async fn initialize_components(cli_catalog: &Catalog) -> Result<(), CLIError> {
+    // Pre-initialization of the singleton is necessary to avoid a situation where,
+    // during the execution of short commands (e.g., `kamu add`),
+    // the service instance is created with a catalog that holds a transaction,
+    // which could potentially lead to a transaction leak.
+    let _flow_service = cli_catalog.get_one::<kamu_flow_system_services::FlowServiceImpl>()?;
+
```

Comment:
I think we can do better.
There are several dependencies like
  - flow service
  - flow configuration service
  - task scheduler
  - task executor
  - ...
which are never activated via CLI commands other than API server or embedded UI.

I think what could be done is to split the catalog with server-only dependencies and CLI base dependencies.
Server could chain the base CLI dependencies catalog.
Once that is  possible, there should not be a problem with the flow service in your scenario.

Reply:
> I think what could be done is to split the catalog with server-only dependencies and CLI base dependencies.

This is a very cool suggestion. This way, in case of sending Outbox Messages, we won't have "unexpected" consumers.